### PR TITLE
CAPG: apidiff prow job added for capg

### PR DESCRIPTION
**Pull request by sayantani11**
This PR adds apidiff prow job for CAPG.
Related to https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/534

### CODE REVIEW

**Review by vincepri — APPROVED** [review]
/ok-to-test
/lgtm

File: config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits-main.yaml [hunk 1]
```diff
@@ -124,6 +124,25 @@ presubmits:
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-gcp
       testgrid-tab-name: pr-e2e-test
+  - name: pull-cluster-api-provider-gcp-apidiff
+    labels:
+      preset-service-account: "true"
```
Comment (cpanato) [inline config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits-main.yaml:129]: I think we don't need a service account for this check

File: config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits-main.yaml [hunk 2]
```diff
@@ -124,6 +124,25 @@ presubmits:
...
+    optional: true
+    decorate: true
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220221-c13e827224-1.23
+        command:
+          - runner.sh
+        args:
+          - ./scripts/ci-apidiff.sh
+        annotations:
```
Comment (cpanato) [inline config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits-main.yaml:143]: this should be in the same ident as the `spec`

File: config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits-release-v1alpha4.yaml [hunk 3]
```diff
@@ -154,3 +154,22 @@ presubmits:
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-gcp
       testgrid-tab-name: pr-conformance-release-v1alpha4
+  - name: pull-cluster-api-provider-gcp-v1alpha4
```
Comment (cpanato) [inline config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits-release-v1alpha4.yaml:157]: please drop this check for this file, we will not backport this change for this branch

File: config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits-release-v1beta1.yaml [hunk 4]
```diff
@@ -225,3 +225,22 @@ presubmits:
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-gcp
       testgrid-tab-name: pr-capi-e2e-release-v1beta1
+  - name: pull-cluster-api-provider-gcp-apidiff-v1beta1
```
Comment (cpanato) [inline config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits-release-v1beta1.yaml:228]: same for this, please drop

**Pull request closed without merging**